### PR TITLE
feature - realm cart

### PR DESCRIPTION
**Pull request by at-lytruong**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@
 //
 
 import UIKit
+import MVVM
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: remove

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 2]
```diff
@@ -97,3 +103,11 @@ extension CartViewController: UITableViewDelegate {
         return 50.0
     }
 }
+
+// MARK: - ViewModelDelegate
+extension CartViewController: ViewModelDelegate {
+    func viewModel(_ viewModel: ViewModel, didChangeItemsAt indexPaths: [IndexPath], for changeType: ChangeType) {
+        tableView.reloadData()
+        priceTotalLabel.text = "\(self.viewModel.caculateTotalPrice()) VND"
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: self.

File: MyApp/View/Controllers/favorite/FavoriteViewController.swift [hunk 3]
```diff
@@ -7,6 +7,7 @@
 //
 
 import UIKit
+import MVVM
```
Comment (at-hieutran) [inline MyApp/View/Controllers/favorite/FavoriteViewController.swift]: remove

File: MyApp/ViewModel/Cart/CartCellViewModel.swift [hunk 4]
```diff
@@ -9,39 +9,54 @@
...
+        guard let cartItem = cartItem else { return }
+        id = cartItem.id
+        image = cartItem.image
+        name = cartItem.name
+        price = cartItem.price
+        quatity = cartItem.quatity
+        currentItem = cartItem
     }
 
     func plus(textField: String) -> String {
```
Comment (at-hieutran) [inline MyApp/ViewModel/Cart/CartCellViewModel.swift]: textField -> đổi tên khác

File: MyApp/ViewModel/Cart/CartCellViewModel.swift [hunk 5]
```diff
@@ -9,39 +9,54 @@
...
 
         guard var quality = Int(textField) else { fatalError("Text empty") }
         quality += 1
         self.quatity = quality
+        update(quality)
         return "\(quality)"
     }
 
     func minus(textField: String) -> String {
```
Comment (at-hieutran) [inline MyApp/ViewModel/Cart/CartCellViewModel.swift]: textField -> Naming

**Comment by at-hieutran** [comment]
LGTM

**Pull request merged**